syntax="proto2";

message Packet_Header {
  
  enum PacketType {
    IQ = 0;
    FFT = 1;
    PDW = 2;
  }
  
  required PacketType type = 1;
  required uint32 fs = 2;
  required uint32 fc = 3;
}

message IQ_Packet {
  repeated float signal = 1;
}


message FFT_Packet {
  repeated double freq_bins_hz = 1;
  repeated double fft = 2;
}

message PDW {
  required double toa_s = 1;
  required double pw_s = 2;
  required double mean_amp_db = 3;
  required double peak_amp_db = 4;
  required double noise_amp_db = 5;
  required double freq_offset_hz = 6;
  repeated float signal = 7;
}

message PDW_Packet {
  repeated PDW pdws = 1;
}

message Packet {
  required Packet_Header header = 1;  
  oneof body {
    IQ_Packet iq_packet = 2;
    FFT_Packet fft_packet = 3;
    PDW_Packet pdw_packet = 4;
  }
}

